lib/src/main/java/com/cloudbees/groovy/cps/CpsTransformer.java

```diff
@@ -1050,25 +1081,23 @@ public void visitDeclarationExpression(final DeclarationExpression exp) {
             makeNode("sequence", new Runnable() {
                 @Override
                 public void run() {
+                    // TODO: Possibly move this logic into multipleAssignment(...) or replace that with more specific
+                    // logic.
                     for (Expression e : exp.getTupleExpression().getExpressions()) {
                         final VariableExpression v = (VariableExpression) e;
                         makeNode("declareVariable", new Runnable() {
                             @Override
                             public void run() {
-                                loc(exp);
                                 literal(v.getType());
                                 literal(v.getName());
                             }
                         });
                     }
-                    makeNode("assign", new Runnable() {
-                        @Override
-                        public void run() {
-                            loc(exp);
-                            visit(exp.getLeftExpression());
-                            visit(exp.getRightExpression());
-                        }
-                    });
+                    if (exp.getRightExpression() instanceof ListExpression) {
+                        multipleAssignment(exp,
+                                exp.getTupleExpression(),
+                                (ListExpression)exp.getRightExpression());
+                    }
```

Comment:
And if not?

Reply:
Again, I should probably add erroring out here, but I'm pretty sure we'd never get that far, i.e., that it'd be invalid syntax caught before we get to this point in compilation.

Reply:
Needs a testcase to prove that tho

Reply:
*nod*

Reply:
Turns out I was wrong about that. `def (a, b) = 4` will error out due to `Integer.getAt(int)` not being a thing, while `def (a, b) = "what"` will result in `a == 'w'` and `b == 'h'`. So I'm rewriting things accordingly.

Reply:
WTF

Reply:
Yeah, I know.